README.md

```diff
@@ -1,2 +1,40 @@
 # terraform-provider-flux
-Terraform provider for Flux v2
+
+This is a Terraform provider for Flux v2, it enables bootstrap a Kubernetes custer with Flux v2 using terraform.
+
+## Example Usage
+
+The provider is consists of two data sources `flux_install` and `flux_sync` the data sources are corresponding to [fluxv2 manifests](https://pkg.go.dev/github.com/fluxcd/flux2@v0.2.1/pkg/manifestgen)
+
+The data sources are returing `YAML` manifest so a second provider is needed to apply the manifest into the Kubernetes cluster. See example folder.
+
+The `flux_install` generates manifests to install the `fluxv2` components.
+
+```hcl
+# Flux
+data "flux_install" "main" {
+  target_path = "staging-cluster"
+  namespace   = "fluxv2"
+}
+```
+
+`flux_sync` the initial source manifest.
+
+```hcl
+data "flux_sync" "main" {
+  target_path = "staging-cluster"
+  url         = "ssh://git@github.com/${var.github_owner}/${var.repository_name}.git"
+}
+```
```

Comment:
I am fine with having an example the README until we get doc generation working with tfplugindocs.

https://github.com/hashicorp/terraform-plugin-docs

Reply:
Are you working on that? You want that into the pipeline?

Reply:
I will get that done tonight, have most of the things in place already.